tests/test_auth.py

```diff
@@ -22,23 +31,23 @@ def test_auth_kwargs():
     assert not auth.authenticate
 
 
-def test_no_credentials_raises(auth_mock):
-    auth_mock.project_id = None
-    auth_mock.project_api_key = None
+def test_no_credentials_raises():
     with pytest.raises(ValueError):
-        auth_mock._find_credentials()
+        Auth()
 
 
-def test_find_credentials_cfg_file(auth_mock):
-    auth_mock.project_id = None
-    auth_mock.project_api_key = None
+def test_cfg_file_not_found():
+    fp = Path(__file__).resolve().parent / "mock_data" / "test_config_fake.json"
+    with pytest.raises(ValueError) as e:
+        Auth(cfg_file=fp)
+    assert "Selected config file does not exist!" in str(e.value)
 
-    fp = Path(__file__).resolve().parent / "mock_data" / "test_config.json"
-    auth_mock.cfg_file = fp
 
-    auth_mock._find_credentials()
-    assert auth_mock.project_id is not None
-    assert auth_mock.project_api_key is not None
+def test_find_credentials_cfg_file():
+    fp = Path(__file__).resolve().parent / "mock_data" / "test_config.json"
+    auth = Auth(cfg_file=fp, authenticate=False)
+    assert auth._credentials_id is not None
+    assert auth._credentials_key is not None
 
```

Comment:
We're missing lots of tests, but most importantly the test of the new authentication method

Reply:
Yes, we agreed to first change the auth class and then test.

Reply:
I have added some initial fixtures and tests for the account_based authentication.

Reply:
Let's avoid such practice in the future:
1. It creates a risk of adding untestable code which will be changed a follow PR anyways
2. It requires certain mental gymnastics to have in check that all changes from an earlier PR are tested in a follow up PR
3. It forces the creation of feature branches instead of merging to the main one in all cases